ftdc/custom_format.go

```diff
@@ -186,43 +280,27 @@ func flatten(datum datum, schema *schema) []float32 {
 			return nil
 		}
 
-		rVal := reflect.ValueOf(stats)
-		if rVal.Kind() == reflect.Pointer {
-			rVal = rVal.Elem()
-		}
-
-		// Use reflection to walk the member fields of an individual set of metric readings. We rely
-		// on reflection always walking fields in the same order.
-		//
-		// Note, while reflection in this way isn't intrinsically expensive, it is making more small
-		// function calls and allocations than some more raw alternatives. For example, we can have
-		// the "schema" keep a (field, offset, type) index and we instead access get a single unsafe
-		// pointer to each structure and walk out index to pull out the relevant numbers.
-		for memberIdx := 0; memberIdx < rVal.NumField(); memberIdx++ {
-			rField := rVal.Field(memberIdx)
-			switch {
-			case rField.CanInt():
-				ret = append(ret, float32(rField.Int()))
-			case rField.CanFloat():
-				ret = append(ret, float32(rField.Float()))
-			default:
-				// Embedded structs? Just grab a global logger for now. A second pass will better
-				// validate inputs/remove limitations. And thread through a proper logger if still
-				// necessary.
-				logging.Global().Warn("Bad number type. Type:", rField.Type())
-				// Ignore via writing a 0 and continue.
-				ret = append(ret, 0)
-			}
+		numbers, err := flattenStruct(reflect.ValueOf(stats))
+		if err != nil {
+			panic(err)
```

Comment:
do we still want this panic here?

Reply:
Good catch, I'll take the time to get rid of it now.